SharedTests/DeferredTests.swift

```diff
@@ -68,4 +68,29 @@ class DeferredTests: XCTestCase {
 
         waitForExpectations(timeout: 10, handler: nil)
     }
+
+    func testAccumulate() {
+        let leak = self.expectation(description: "deinit")
+
+        class TestClass {
+            let end: XCTestExpectation
+            init(e: XCTestExpectation) {
+                end = e
+                accumulate([self.aSimpleFunction]).upon { _ in
+
+                }
+            }
+
+            func aSimpleFunction() -> Success {
+                return succeed()
+            }
+            deinit {
```

Comment:
Does this exercise the bug because `TestClass` won't be deinited if one of its methods is alive as a thunk?

If so, `weak self` in `aSimpleFunction` would make this test pass if the bug were still present, right?

Is there a way to write a test that really exercises the leak? Naturally you can't have a weak reference to an array, so we can't check it that way…

You also need a little test for what happens if one of the thunks returns failure.